Sources/Differ/Diff+AppKit.swift

```diff
@@ -1,31 +1,86 @@
 #if canImport(AppKit) && !targetEnvironment(macCatalyst)
 import AppKit
 
+// MARK: - NSTableView
+
 extension NSTableView {
 
+    /// Animates rows which changed between oldData and newData using custom `isEqual`.
+    ///
+    /// - Parameters:
+    ///   - oldData:            Data which reflects the previous state of `NSTableView`
+    ///   - newData:            Data which reflects the current state of `NSTableView`
+    ///   - isEqual:            A function comparing two elements of `T`
+    ///   - deletionAnimation:  Animation type for deletions
+    ///   - insertionAnimation: Animation type for insertions
+    ///   - rowIndexTransform:  Closure which transforms a zero-based row to the desired index
+    public func animateRowChanges<T: Collection>(
+        oldData: T,
+        newData: T,
+        isEqual: EqualityChecker<T>,
+        deletionAnimation: NSTableView.AnimationOptions = [],
+        insertionAnimation: NSTableView.AnimationOptions = [],
+        rowIndexTransform: (Int) -> Int = { $0 }
+    ) {
+        apply(
+            oldData.extendedDiff(newData, isEqual: isEqual).patch(from: oldData, to: newData),
+            deletionAnimation: deletionAnimation,
+            insertionAnimation: insertionAnimation,
+            rowIndexTransform: rowIndexTransform
+        )
+    }
+
     /// Animates rows which changed between oldData and newData.
     ///
     /// - Parameters:
     ///   - oldData:            Data which reflects the previous state of `NSTableView`
     ///   - newData:            Data which reflects the current state of `NSTableView`
     ///   - deletionAnimation:  Animation type for deletions
     ///   - insertionAnimation: Animation type for insertions
-    ///   - indexPathTransform: Closure which transforms zero-based `IndexPath` to desired `IndexPath`
+    ///   - rowIndexTransform:  Closure which transforms a zero-based row to the desired index
     public func animateRowChanges<T: Collection>(
         oldData: T,
         newData: T,
         deletionAnimation: NSTableView.AnimationOptions = [],
         insertionAnimation: NSTableView.AnimationOptions = [],
-        indexPathTransform: (IndexPath) -> IndexPath = { $0 }
+        rowIndexTransform: (Int) -> Int = { $0 }
     ) where T.Element: Equatable {
         apply(
-            oldData.extendedDiff(newData),
+            extendedPatch(from: oldData, to: newData),
```

Comment:
Not that I'm against this change, but why the move to using a pure function? Just for cleanliness/clarity of use?

Reply:
Ah, no strong opinion involved in this. I'm just new to the codebase and found this the shortest way to create the patches. Apart from this function I'm only aware of writing inline what the function does, like `oldData.extendedDiff(newData).patch(from:oldData,to:newData)`, is there another way you would recommend?

Reply:
No, that's completely fine - I tend to lean into methods more than free functions, so I was curious what your thinking was 👍